rspec-core/lib/rspec/core/formatters/exception_presenter.rb

```diff
@@ -184,13 +184,10 @@ def failure_slash_error_lines
 
         # rubocop:disable Lint/RescueException
         # :nocov:
-        if SyntaxError.instance_methods.include?(:detailed_message)
+        if RSpec::Support::RubyFeatures.supports_exception_detailed_message?
           def exception_message_string(exception)
-            case exception
-            when SyntaxError then exception.detailed_message.to_s
-            else
-              exception.message.to_s
-            end
+            puts "exception.class: #{exception.class}"
+            exception.detailed_message(:highlight => false).sub(/\s\(#{Regexp.escape(exception.class.to_s)}\)$/, '')
```

Comment:
I was concerned about this hash being converted into keyword arguments, which Ruby 2.7 removed, but no warnings were raised.